components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -4908,6 +4914,77 @@ public String getExternalWorkflowReferenceId(int subscriptionId) throws APIManag
         return apiMgtDAO.getExternalWorkflowReferenceForSubscription(subscriptionId);
     }
 
+    @Override
+    public int addCertificate(String userName, String certificate, String alias, String endpoint)
+            throws APIManagementException {
+        ResponseCode responseCode = ResponseCode.INTERNAL_SERVER_ERROR;
+        CertificateManager certificateManager = new CertificateManagerImpl();
+        String tenantDomain = MultitenantUtils.getTenantDomain(userName);
+        ;
+        try {
+            int tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                    .getTenantId(tenantDomain);
+            responseCode = certificateManager
+                    .addCertificateToPublisher(certificate, alias, endpoint, tenantId);
+
+            if (responseCode == ResponseCode.SUCCESS) {
+                //Get the gateway manager and add the certificate to gateways.
+                GatewayCertificateManager gatewayCertificateManager = new GatewayCertificateManager();
+                gatewayCertificateManager.addToGateways(certificate, alias);
+            } else {
+                log.error("Adding certificate to the Publisher node is failed. No certificate changes will be " +
+                        "affected.");
+            }
+        } catch (UserStoreException e) {
+            throw new APIManagementException("Error while reading tenant information", e);
+        }
+        return responseCode.getResponseCode();
+    }
+
+    @Override
+    public int deleteCertificate(String userName, String alias, String endpoint) throws APIManagementException {
+        ResponseCode responseCode = ResponseCode.INTERNAL_SERVER_ERROR;
+        CertificateManager certificateManager = new CertificateManagerImpl();
+        String tenantDomain = MultitenantUtils.getTenantDomain(userName);
+
+        try {
+            int tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                    .getTenantId(tenantDomain);
+            responseCode = certificateManager.deleteCertificateFromPublisher(alias, endpoint, tenantId);
```

Comment:
As I mention earlier remove the publisher node erc and make method more meaningful